File: Projeto-TCC/ACS-Cadastro-Web/persistences/RecordDataPersistence.php
<?php

class RecordDataPersistence
{

    static function insert($recordDetails, $anthropometric, $kidAndPregnant)
    {
        $db = new AcsDataBase(AcsDataBase::DB_NAME);;
        $idDetails = self::insertRecordDetails($db, $recordDetails);
        $idAnthropometric = self::insertAnthropometric($db, $anthropometric);
        $idKidAndPregnant = self::insertKidAndPregnant($db, $kidAndPregnant);

        $query = "INSERT INTO TB_RECORD_DATA (ID_RECORD_DETAILS, ID_ANTROPOMETRIC, ID_KID_PREGNANT)
                  VALUES(:ID_RECORD_DETAILS, :ID_ANTROPOMETRIC, :ID_KID_PREGNANT)";
        $array = array(":ID_RECORD_DETAILS" => $idDetails,
                        ":ID_ANTHROPOMETRIC" => $idAnthropometric,
                        ":ID_KID_PREGNANT" => $idKidAndPregnant);
        $id = $db->insert($query, $array);

        if(!$id){
            echo "Erro ao inserir o registro";
            die();
        }

        return $id;
    }

    private static function insertRecordDetails($db, $recordDetails)
    {
        //$numSus = $recordDetails[':NUM_SUS'];
        //$idCitizen = AgentsPersistence::get($numSus);
        $idCitizen = 200;
        $recordDetails[":ID_CITIZEN"] = $idCitizen  ;
        unset($recordDetails[':NUM_SUS']);

        $query = "INSERT INTO  TB_RECORD_DETAILS (RECORD, PLACE_CARE, TYPE_CARE, SHIFT, ID_CITIZEN)
                  VALUES (:RECORD, :PLACE_CARE, :TYPE_CARE, :SHIFT, :ID_CITIZEN)";

        $id = $db->insert($query, $recordDetails);

        if(!$id){
            echo "Erro ao inseriro registro";
            die();
        }
        return $id;
    }

    private static function insertAnthropometric($db, $anthropometric)
    {
        $query = "INSERT INTO TB_ANTROPOMETRIC (WEIGHT, HEIGHT, VACCINATES)
                  VALUES (:WEIGHT, :HEIGHT, :VACCINATES)";

        var_dump($anthropometric); echo "<br><br>";

        $id = $db->insert($query, $anthropometric);
        if(!$id){
            echo "Erro ao inseriro registro";
            die();
        }
        return $id;
    }

    private static function insertKidAndPregnant($db, $kidAndPregnant)
    {
        $query = "INSERT INTO TB_KID_PREGNANT (BREAST_FEEDING, DUM, PLANNED_PREGNANCY, WEEKS, PREVIOUS, CHILD_BIRTH, HOME_CARE)
                  VALUES (:BREAST_FEEDING, :DUM, :PLANNED_PREGNANCY, :WEEKS, :PREVIOUS, :CHILD_BIRTH, :HOME_CARE)";

        var_dump($kidAndPregnant); echo "<br><br>";

        $id = $db->insert($query, $kidAndPregnant);
        if(!$id){
            echo "Erro ao inseriro registro";
            die();
        }
        return $id;
    }
}

File: Projeto-TCC/ACS-Cadastro-Mobile/app/src/main/java/tcc/acs_cadastro_mobile/persistence/RecordDataPersistence.java
package tcc.acs_cadastro_mobile.persistence;

import io.realm.Realm;
import tcc.acs_cadastro_mobile.models.CitizenModel;
import tcc.acs_cadastro_mobile.models.RecordDataModel;
import tcc.acs_cadastro_mobile.subModels.Anthropometric;
import tcc.acs_cadastro_mobile.subModels.KidAndPregnant;
import tcc.acs_cadastro_mobile.subModels.RecordDetails;

public class RecordDataPersistence {

    private RecordDataPersistence() {}

    public static RecordDataModel get(RecordDetails details, Anthropometric anthropometric, KidAndPregnant kidAndPregnant){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        RecordDataModel object = realm.copyToRealm(new RecordDataModel(details, anthropometric, kidAndPregnant));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static RecordDetails get(long record, String placeCare, String typeCare, String shift, CitizenModel citizen){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        RecordDetails object = realm.copyToRealm(new RecordDetails(record, placeCare, typeCare, shift, citizen));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static Anthropometric get(float weight, int height, boolean vaccinates){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        Anthropometric object = realm.copyToRealm(new Anthropometric(weight, height, vaccinates));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static KidAndPregnant get(String breastFeeding, String dum, boolean plannedPregnancy, int weeks,
                                                  int previous, int childBirth, String homeCare){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        KidAndPregnant object = realm.copyToRealm(new KidAndPregnant(breastFeeding, dum, plannedPregnancy,
                weeks, previous, childBirth, homeCare));
        realm.commitTransaction();
        realm.close();
        return object;
    }
}
